src/features/star-pair/storage.ts

```diff
@@ -0,0 +1,35 @@
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+
+export interface Pair {
+  base: Metadata;
+  quote: Metadata;
+}
+
+const STAR_STORE_LS_KEY = 'star-pairs-store';
+
+export const getStarredPairs = (): Pair[] => {
+  try {
+    const data = JSON.parse(localStorage.getItem(STAR_STORE_LS_KEY) ?? '[]') as {
+      base: string;
+      quote: string;
+    }[];
+    return data.map(pair => ({
+      base: Metadata.fromJson(pair.base),
+      quote: Metadata.fromJson(pair.quote),
+    }));
+  } catch (_) {
+    return [];
+  }
+};
+
+export const setStarredPairs = (pairs: Pair[]): void => {
+  localStorage.setItem(
+    STAR_STORE_LS_KEY,
+    JSON.stringify(
+      pairs.map(pair => ({
+        base: pair.base.toJson(),
+        quote: pair.quote.toJson(),
+      })),
```

Comment:
question: would it be easier if we just stored symbol pairs versus the whole metadata objects?

Reply:
this would delay showing data to user. Idk, for me it's more complicated to calculate the Metadata from symbol than storing it like this